Makefile

```diff
@@ -184,7 +184,7 @@ endif
 
 
 # Generate dependency information
-CFLAGS += -MMD -MP -MF"$(@:%.o=%.d)"
+CFLAGS += -MMD -MP -MF"$(@:%.o=%.d)" -DGIT_HASH="$(shell git rev-parse HEAD)"
```

Comment:
Am I correct in understanding that this would create a constant variable accessible to the C code which is the latest commit hash?

Reply:
How about windows users compiling through TrueStudio?

Reply:
That's correct; this is equivalent to a #define. As noted in the PR, I have very little experience with TrueStudio, so I'm not sure how to implement this - someone more familiar with its build config system will have to finish the implementation on that end. This will not prevent compilation; info.h contains a failsafe value.